contrib/jackc/pgx.v5/pgx_tracer.go

```diff
@@ -175,6 +183,38 @@ func (t *pgxTracer) TraceConnectEnd(ctx context.Context, data pgx.TraceConnectEn
 	finishSpan(ctx, data.Err)
 }
 
+func (t *pgxTracer) TraceAcquireStart(ctx context.Context, pool *pgxpool.Pool, _ pgxpool.TraceAcquireStartData) context.Context {
+	if !t.cfg.traceAcquire {
+		return ctx
+	}
+	opts := t.spanOptions(pool.Config().ConnConfig, operationTypeAcquire, "")
+	_, ctx = tracer.StartSpanFromContext(ctx, "pgx.acquire", opts...)
+	return ctx
+}
+
+func (t *pgxTracer) TraceAcquireEnd(ctx context.Context, _ *pgxpool.Pool, data pgxpool.TraceAcquireEndData) {
+	data.Conn.PgConn().CustomData()[customDataContextKey] = ctx
+	if !t.cfg.traceAcquire {
+		return
+	}
+	finishSpan(ctx, data.Err)
+}
+
+func (t *pgxTracer) TraceRelease(pool *pgxpool.Pool, data pgxpool.TraceReleaseData) {
+	if !t.cfg.traceRelease {
+		return
+	}
+	acquireCtx, ok := data.Conn.PgConn().CustomData()[customDataContextKey].(context.Context)
+	opts := t.spanOptions(pool.Config().ConnConfig, operationTypeRelease, "")
+	if ok {
+		_, ctx := tracer.StartSpanFromContext(acquireCtx, "pgx.release", opts...)
+		finishSpan(ctx, nil)
+	} else {
+		span := tracer.StartSpan("pgx.release", opts...)
+		span.Finish()
+	}
```

Comment:
Is having a `pgx.release` span with a duration of almost `0` the right thing here? Wouldn't it be better if `TraceAcquireEnd` would start a `pgx.hold` or span or similar and then end this span here? This way you could easily track how long a pool is being held during after it's been acquired.

Reply:
yeah agreed this is the better way, updated